### PR TITLE
feat: process stdio streams for Deno

### PR DESCRIPTION
**Pull request by Mesteery**
Fixes https://github.com/jspm/jspm-core/issues/10.

*With Deno it works. In the browser, I have not tested.*

### CODE REVIEW

**Review by guybedford — COMMENTED** [review]
One thing that could help here is trying to see if it's possible to modify the first data event to internally lazily load the streams implementation then pipe through the first data event and do the right kind of stitching to get the event system to behave normally giving the normal API while doing an invisible async lazy load in the background via dynamic import.

For Deno the benefit would be not needing higher level permissions when only using process without these streams.

For the browser the benefit is that process is such a highly used implementation that it would save unnecessarily loading the full streams API for lots of apps that don't need it.

This isn't straightforward I know, but is likely needed to ensure the right level of user experience for this one.

**Comment by Mesteery** [comment]
> For Deno the benefit would be not needing higher level permissions when only using process without these streams.

Currently, importing streams doesn't require higher level permissions, so lazy-loading streams in Deno may not be necessary.

For browsers, maybe we can just change exports?

**Comment by guybedford** [comment]
Ah right it’s just accessing the stdin and stdout that requires higher
permissions but that is still lazy anyway.

For the browser case it is important to come up with a lazy streams
implementation solution if this is implemented. Alternatively we can just
leave out the browser case entirely for now.

**Comment by Mesteery** [comment]
> For the browser case it is important to come up with a lazy streams
implementation solution if this is implemented. Alternatively we can just
leave out the browser case entirely for now.

Yes I agree, but if we change the exports, we can just use a getter and lazy-load.

**Comment by guybedford** [comment]
You can’t make the getter async though, hence the need to inline it into
the stream itself.

**Comment by Mesteery** [comment]
Indeed, I didn't think about it.

File: nodelibs/deno/process.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
-import { notImplemented } from 'https://deno.land/std@0.103.0/node/_utils.ts';
-import { fileURLToPath } from 'https://deno.land/std@0.103.0/node/url.ts';
+import { notImplemented } from 'https://deno.land/std@0.105.0/node/_utils.ts';
```
Comment (guybedford) [inline nodelibs/deno/process.ts:2]: Also if updating to std@0.105.0, we should do this in all Deno files.

**Review by guybedford — APPROVED** [review]

**Pull request merged**